scripts/update_robot.py

```diff
@@ -183,7 +184,55 @@ def on_shutdown():
     return nl.rc
 
 
+def rethink_ssh_update_version():
+    """
+    Verifies the version of the software is older than 1.1.0.
+    If newer, print out url to SSH update instructions
+    and return False
+    """
+    param_name = "rethink/software_version"
+    robot_version = rospy.get_param(param_name, None)
+    if not robot_version:
+        rospy.logerr("RobotUpdater: Failed to retrieve robot version "
+                     "from rosparam: %s\n"
+                     "Verify robot state and connectivity "
+                     "(i.e. ROS_MASTER_URI)", param_name)
+        return False
+    else:
+        # parse out first 3 digits of robot version tag
+        pattern = ("^([0-9]+)\.([0-9]+)\.([0-9]+)")
+        match = re.search(pattern, robot_version)
+        if not match:
+	    rospy.logwarn("RobotEnable: Invalid robot version: %s",
+                robot_version)
+	    return True
+        robot_version = match.string[match.start(1):match.end(3)]
+        def detect_ssh_update_required(robot_version):
+	    # Returns True if the robot's version is older
+	    # than RSDK 1.1.0
+	    v_list = robot_version.split('.')
+	    if float(v_list[0]) < 1:
+	        return False
+	    elif float(v_list[0]) == 1 and float(v_list[1]) < 1:
+	        return False
+	    else:
+	        return True
+        if detect_ssh_update_required(robot_version):
+	    errstr_version = ("RobotUpdater: Your Baxter's software "
+                "version is (%s).\nFor robots with software 1.1.0 and newer, "
+	        "software updates must be run by SSH'ing into the robot.\n"
+	        "Please see "
+	        "http://sdk.rethinkrobotics.com/wiki/SSH_Update for a "
+	        "detailed tutorial on updating your robot over SSH.") 
+            rospy.logerr(errstr_version, robot_version)
+            return False
+    return True
+
+
 def main():
+    rospy.init_node('rsdk_update_robot')
```

Comment:
Hmm. I think you're right about `rosparam`. However, `rospy.logerr` definitely needs the `rospy.init_node` first, otherwise:
```
No handlers could be found for logger "rosout"
```